packages/isar_core/src/native/isar_deserializer.rs

```diff
@@ -185,3 +185,779 @@ impl<'a> IsarDeserializer<'a> {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::IsarDeserializer;
+    use super::{NULL_BOOL, NULL_BYTE, NULL_DOUBLE, NULL_FLOAT, NULL_INT, NULL_LONG};
+    use crate::core::data_type::DataType;
+    use byteorder::{ByteOrder, LittleEndian};
+
+    static LOREM: &str = "Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat. Duis aute irure dolor in reprehenderit in voluptate velit esse cillum dolore eu fugiat nulla pariatur. Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum. Sed ut perspiciatis unde omnis iste natus error sit voluptatem accusantium doloremque laudantium, totam rem aperiam, eaque ipsa quae ab illo inventore veritatis et quasi architecto beatae vitae dicta sunt explicabo. Nemo enim ipsam voluptatem quia voluptas sit aspernatur aut odit aut fugit, sed quia consequuntur magni dolores eos qui ratione voluptatem sequi nesciunt. Neque porro quisquam est, qui dolorem ipsum quia dolor sit amet, consectetur, adipisci velit, sed quia non numquam eius modi tempora incidunt ut labore et dolore magnam aliquam quaerat voluptatem. Ut enim ad minima veniam, quis nostrum exercitationem ullam corporis suscipit laboriosam, nisi ut aliquid ex ea commodi consequatur? Quis autem vel eum iure reprehenderit qui in ea voluptate velit esse quam nihil molestiae consequatur, vel illum qui dolorem eum fugiat quo voluptas nulla pariatur?";
+
+    #[test]
+    fn test_is_null() {
+        let bytes = &[
+            NULL_BYTE,
+            NULL_BOOL,
+            (NULL_INT & 0xFF) as u8,
+            ((NULL_INT >> 8) & 0xFF) as u8,
+            ((NULL_INT >> 16) & 0xFF) as u8,
+            ((NULL_INT >> 24) & 0xFF) as u8,
+            (NULL_LONG & 0xFF) as u8,
+            ((NULL_LONG >> 8) & 0xFF) as u8,
+        ];
```

Comment:
Maybe I am mistaken, but shouldn't this contain a header? Otherwise the `NULL_BYTE`, `NULL_BOOL` and the first byte of `NULL_INT` will be used as the header?

Reply:
you are right, don't know what happened there, updated the tests